pkgs/development/python-modules/xontrib-abbrevs/default.nix

```diff
@@ -0,0 +1,48 @@
+{
+  buildPythonPackage,
+  lib,
+  fetchFromGitHub,
+
+  poetry-core,
+  prompt-toolkit,
+  pytest,
+  xonsh
+}:
+
+buildPythonPackage rec {
+  name = "xontrib-abbrevs";
+  version = "0.0.1";
+
+  src = fetchFromGitHub {
+    owner = "xonsh";
+    repo = name;
+    rev = version;
+    sha256 = "sha256-DrZRIU5mzu8RUzm0jak/Eo12wbvWYusJpmqgIAVwe00=";
+  };
+
+  format = "pyproject";
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    prompt-toolkit
+  ];
+
+  patches = [
+    ./remove-xonsh-dependency.patch
+  ];
+
+  checkInputs = [
+    pytest
+    xonsh
+  ];
+
+  meta = with lib; {
+    description = "Command abbreviations. This expands input words as you type.";
+    homepage = "https://github.com/xonsh/${name}";
```

Comment:
Developers tend to open this link from within their editor, so a substitution here is not a good idea.